src/libYARP_OS/src/Node.cpp

```diff
@@ -350,6 +436,10 @@ void yarp::os::Node::Helper::add(Contactable& contactable)
     }
     prepare(name);
     item.contactable = &contactable;
+    ROSReport* rep = new ROSReport();
+    item.contactable->setReporter(*rep);
```

Comment:
@Tobias-Fischer Unfortunately, after investigating a little bit, this works and it will mess up with the node:
```
    sub = new yarp::os::Subscriber<std_msgs::String>();
    if (!sub->topic("/yarp/foo")) {
        yError() << "Failed to create subscriber to /yarp/foo\n";
        return -1;
    }
    MyReport report;
    sub->asPort().setReporter(report);
```